synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -646,7 +646,11 @@ def get_rejected_events(
             if has_state and has_event_auth:
                 continue
 
-            room_version_obj = KNOWN_ROOM_VERSIONS[room_version]
+            room_version_obj = KNOWN_ROOM_VERSIONS.get(room_version)
+            if not room_version_obj:
+                # We no longer suppport this room version, so we just ignore the
```

Comment:
```suggestion
                # We no longer support this room version, so we just ignore the
```

Reply:
Also, maybe we should at least log the room ID?

Reply:
We don't easily have access to the room ID, so I've just logged the event ID + room version